lib/plug/router.ex

```diff
@@ -446,7 +446,7 @@ defmodule Plug.Router do
   defmacro forward(path, options) do
     quote bind_quoted: [path: path, options: options] do
       {target, options} = Keyword.pop(options, :to)
-      {options, plug_options} = Keyword.split(options, [:host, :private, :assigns])
+      {options, plug_options} = Keyword.split(options, [:via, :host, :private, :assigns])
```

Comment:
Yes indeed that could be the case. I tend to think that forwarding "other remaining" options to the Plug is not such a good practice anyways when you have the `:init_opts` option to `forward/2` exactly for that 😄 .

In general, that behavior prevents us from adding any options to `forward/2`, so I personally would really love to see it deprecated. Thoughts @josevalim?

Reply:
Yeah, `init_opts` is preferred but I am not sure about deprecations because the API rarely changes anyway. We should mention in the docs though `init_opts` is preferred.

Reply:
@josevalim I've added the test and a line in the docstring about init_opts being preferred